skyportal/app_server.py

```diff
@@ -219,9 +223,18 @@ def make_app(cfg, baselayer_handlers, baselayer_settings):
 
     app = tornado.web.Application(handlers, **settings)
     models.init_db(**cfg['database'])
-    baselayer_model_util.create_tables()
-    model_util.refresh_enums()
-    model_util.setup_permissions()
+
+    # Only modify tables in one application instance
+    if process == 0:
+
+        # If tables are found in the database, new tables will only be added
+        # in debug mode.  In production, we leave the tables alone, since
+        # migrations might be used.
+        baselayer_model_util.create_tables(add=env.debug)
+
+        model_util.refresh_enums()
+        model_util.setup_permissions()
```

Comment:
nice

Reply:
Turned out not to work; I still don't understand why not!